packages/bot-web-ui/src/pages/bot-builder/quick-strategy/inputs/__tests__/qs-input.spec.tsx

```diff
@@ -92,4 +88,69 @@ describe('<QSInput />', () => {
 
         expect(container_element).not.toHaveClass('no-focus');
     });
+
+    it('the onMouseEnter handler sets the focus to false if the error or focus on element is absent ', () => {
+        mock_store.ui.is_mobile = false;
+        render(<QSInput name='duration' type='number' min={100} onChange={onChange} />, {
+            wrapper,
+        });
+
+        const container_element = screen.getByTestId('dt_qs_input_container');
+        userEvent.type(container_element, '{enter}');
+
+        expect(container_element).not.toHaveClass('no-focus');
+    });
+
+    it('the decrease should round a decimal number to two decimal places', () => {
+        render(<QSInput name='size' type='number' onChange={onChange} />, {
+            wrapper,
+        });
+
+        const btn_element = screen.getByTestId('dt_qs_input_decrease');
+        userEvent.click(btn_element);
+
+        const input_decrease = screen.getByTestId('dt_qs_size') as HTMLInputElement;
+        const input_value = input_decrease.value;
+        expect(input_value).toEqual('4.18');
+    });
+
+    it('the increase icon should round a decimal number to two decimal places', () => {
+        render(<QSInput name='size' type='number' max={7} onChange={onChange} />, {
+            wrapper,
+        });
+
+        const btn_element = screen.getByTestId('dt_qs_input_increase');
+        userEvent.click(btn_element);
+
+        const input_decrease = screen.getByTestId('dt_qs_size') as HTMLInputElement;
+        const input_value = input_decrease.value;
+        expect(input_value).toEqual('6.18');
+    });
+
+    it('should be able to call handleOnChange() when type equals number and display the same number', () => {
+        render(<QSInput name='duration' type='number' onChange={onChange} />, {
+            wrapper,
+        });
+        const input_element = screen.getByTestId('dt_qs_duration');
+        userEvent.type(input_element, '1');
+        expect(input_element).toHaveDisplayValue(['1']);
+    });
+
+    it('should be able to call handleOnChange() when type equals text by default and display default number', () => {
+        render(<QSInput name='duration' onChange={onChange} />, {
+            wrapper,
+        });
+        const input_element = screen.getByTestId('dt_qs_duration');
+        userEvent.type(input_element, 'a');
+        expect(input_element).toHaveDisplayValue(['1']);
+    });
+
+    it('should display a number of not more than 12 digits and be able to call handleOnChange() when type equals number, max length equals or more than 12', () => {
+        render(<QSInput name='loss' type='number' onChange={onChange} />, {
+            wrapper,
+        });
+        const input_element = screen.getByTestId('dt_qs_loss');
+        userEvent.type(input_element, '9');
+        expect(input_element).toHaveDisplayValue(['999999999999']);
+    });
```

Comment:
@maryia-matskevich-deriv I need a small clarification here, after inputting just '9' how are we expecting '999999999999'?

Reply:
@shafin-deriv initial loss is 999999999999, when you type '9', it makes 99999999999**9**
```
        const initial_value = {
            duration: 1,
            size: 5.1789,
            loss: 999999999999,
        };
```